blog/_posts/2021-11-15-k8s-custom-deploy.md

```diff
@@ -0,0 +1,106 @@
+When using [`k8s_custom_deploy`][api-k8s_custom_deploy], Tilt features like status reporting and log streaming work automatically.
+Even the more advanced customizations provided by [`k8s_resource`][api-k8s_resource] such as port forwards are supported!
+
+## Example
+Let's use [`k8s_custom_deploy`][api-k8s_custom_deploy] to deploy Kafka using a public Helm chart into our cluster:
+```python
+# configure the remote Helm repo
+local('helm repo add bitnami https://charts.bitnami.com/bitnami',
+      quiet=True,
+      echo_off=True)
+
+# on deploy -> run the Helm install and then query for what it deployed
+apply_cmd = """
+helm upgrade --install -f values.yaml local-kafka bitnami/kafka 1>&2
+helm get manifest local-kafka | kubectl get -oyaml -f -
+"""
+
+# on `tilt down` -> uninstall the release with Helm
+delete_cmd = 'helm uninstall local-kafka'
+
+k8s_custom_deploy(
+    'kafka',
+    apply_cmd=apply_cmd,
+    delete_cmd=delete_cmd,
+    # apply_cmd will be re-executed whenever these files change
+    deps=['values.yaml'],
+)
+
+# add a port forward so it's possible to access/debug the cluster from
+# our host machine
+k8s_resource('kafka', port_forwards=['9092:9092'])
+```
+
+Here's what it looks like when we run `tilt up`:
+![Tilt web UI after running "tilt up" against the sample Tiltfile included in this blog post](/assets/images/k8s-custom-deploy/demo.gif)
+
+Tilt runs the `apply_cmd` passed to [`k8s_custom_deploy`][api-k8s_custom_deploy] on `tilt up` and whenever any path from the `deps` argument changes.
+The `apply_cmd` deploys to Kubernetes (`helm upgrade --install ...`) and writes the _result_ of the deploy as YAML to stdout (`helm get manifest...`).
+(This is what allows Tilt to know what was deployed so that it can monitor status, tail logs, set up port forwards, and perform Live Updates.
+As a result, all logs for the deploy are redirected to stderr with `1>&2` so that _only_ the YAML is written to stdout.
+For more details, see the [Tiltfile API reference][api-k8s_custom_deploy].)
+
+If we query Helm, we'll see that it knows about this release:
+```bash
+$ helm ls
+NAME       	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART       	APP VERSION
+local-kafka	default  	3       	2021-11-16 09:24:48.386254 -0500 EST	deployed	kafka-14.4.1	2.8.1
+```
+
+Now, if we run `tilt down`, we'll see that our `delete_cmd` is invoked:
+```bash
+$ tilt down
+Loading Tiltfile at: ./Tiltfile
+Successfully loaded Tiltfile (177.134ms)
+Running cmd: helm uninstall local-kafka
+release "local-kafka" uninstalled
+```
+This allows tools like Helm to not only delete any entities from Kubernetes that they created but also clean up any extra state (e.g. Helm release metadata).
+
+## What's The Catch?
+We are still exploring the best semantics for passing Tilt-built image references to external tools, so they can interoperate with `docker_build`.
```

Comment:
I feel like the problem here is not clear enough. I'd recommend adding a sentence about it.
"At the moment, it is still not possible to build an image externally and pass that image reference into the k8s_custom_deploy function"